### ISSUE TITLE
Broken TLS tests in master branch

### ISSUE THREAD

**Issue by BillyYccc** [eclipse-vertx/vertx-sql-client#378]
the TLS tests in PostgreSQL do not pass due to the breaking changes of `upgradeToSsl` in Vert.x core 4.0. We need a rework implementation for TLS in PostgreSQL and that also makes sense for MySQL TLS support in #299.

**Comment by vietj** [comment]
ah right

**Comment by vietj** [comment]
tests currently pass, so I'm not sure what you are refering to

**Comment by vietj** [comment]
currently I think we still have tests passing:

```
        conn.socket().upgradeToSsl(v -> {
          ctx.pipeline().remove(this);
          upgradePromise.complete();
        });
```

but that does not look correct since now v is an AsyncResult handler.

**Comment by BillyYccc** [comment]
The travis CI does not pass, see https://travis-ci.org/eclipse-vertx/vertx-sql-client/builds/563626290?utm_source=github_status&utm_medium=notification.

**Comment by BillyYccc** [comment]
I think we should refactor the `InitiateSslHandler` class to avoid exception capature and use the `upgradeToSsl` callback AsyncResult.

**Comment by vietj** [comment]
weird, it works locally for me

**Comment by vietj** [comment]
it's maybe a race

**Comment by vietj** [comment]
I might have a (temporary) workaround for this

**Comment by vietj** [comment]
this is related to a bug in vertx-core actually where the upgradeToSSL test will not setup the current Vert.x context association correctly and thus the socket connection does not run on the context

**Comment by vietj** [comment]
going to push a fix in vertx-core for this first

**Comment by vietj** [comment]
see https://github.com/eclipse-vertx/vert.x/commit/6a255436777a77640b8f31da35ef96876d773578

**Comment by vietj** [comment]
I patched 3.8 and master with work around that will make TLS work in both cases
